### PR TITLE
Replace GMaps dependency with PyCountry

### PR DESCRIPTION
**Pull request by Alchez**
Also handles errors found in https://github.com/DigiThinkIT/jhaudio_customizations/issues/753.

### CODE REVIEW

File: erpnext_taxjar/api.py [hunk 1]
```diff
@@ -89,38 +99,65 @@ def create_transaction(doc, method):
...
+
+	if company_address:
+		if doc.shipping_address_name:
+			shipping_address = frappe.get_doc("Address", doc.shipping_address_name)
+		else:
+			shipping_address = company_address
+	else:
+		return
+
+	if not shipping_address.country == "United States":
```
Comment (vjFaLk) [inline erpnext_taxjar/api.py:125]: Move this to the top, this whole thing is redundant if the address isn't in the US

File: erpnext_taxjar/api.py [hunk 2]
```diff
@@ -89,38 +99,65 @@ def create_transaction(doc, method):
...
+	if address_state in us_states:
+		shipping_state = address_state
+	else:
+		try:
+			state = pycountry.subdivisions.lookup(address_state)
+		except LookupError:
+			error_message = """{} is not a valid state!
+							Check for typos or enter the
+							2-digit code for your state
+							found <a href="https://www.ups.com/worldshiphelp/WS16/ENU/AppHelp/Codes/State_Province_Codes.htm">
```
Comment (vjFaLk) [inline erpnext_taxjar/api.py]: Replace with non-commercial link, maybe link Wiki

File: erpnext_taxjar/api.py (hunk 1 above)
Comment (Alchez) [reply erpnext_taxjar/api.py:125]: This is the furthest I could move the validation.

File: erpnext_taxjar/api.py [hunk 3]
```diff
@@ -89,38 +99,65 @@ def create_transaction(doc, method):
...
+
+	if address_state in us_states:
+		shipping_state = address_state
+	else:
+		try:
+			state = pycountry.subdivisions.lookup(address_state)
+		except LookupError:
+			error_message = """{} is not a valid state!
+							Check for typos or enter the
+							2-digit code for your state
```
Comment (vjFaLk) [inline erpnext_taxjar/api.py]: Two letter

File: erpnext_taxjar/api.py [hunk 4]
```diff
@@ -89,38 +99,65 @@ def create_transaction(doc, method):
...
+		else:
+			shipping_address = company_address
+	else:
+		return
+
+	if not shipping_address.country == "United States":
+		return
+
+	us_states = pycountry.subdivisions.get(country_code='US')
+	us_states = [state.code.split('-')[1] for state in us_states]
```
Comment (vjFaLk) [inline erpnext_taxjar/api.py:129]: Remove this entirely

**Review by vjFaLk — APPROVED** [review]

**Comment by vjFaLk** [comment]
Add screenshot for errors

**Comment by Alchez** [comment]
Screenshots:

**1. TaxJar error handling:**
image

**2. PyCountry error handling:**
image

File: erpnext_taxjar/api.py (hunk 4 above)
Comment (Alchez) [reply erpnext_taxjar/api.py:129]: ISO Codes (for the US atleast) use the Country Code as the prefix to the state code. For example, FL becomes US-FL. If I use a simple lookup with FL as the state in the form, pycountry will catch a false negative error.

**Pull request merged**